Nos/Service/Relay/RelaySubscriptionManager.swift

```diff
@@ -166,15 +169,51 @@ actor RelaySubscriptionManagerActor: RelaySubscriptionManager {
     // MARK: - Talking to Relays
     
     func processSubscriptionQueue() async {
-        let waitingLongSubscriptions = all.filter { !$0.isOneTime && !$0.isActive }
-        let waitingOneTimeSubscriptions = all.filter { $0.isOneTime && !$0.isActive }
-        
-        waitingOneTimeSubscriptions.forEach { start(subscription: $0) }
-        waitingLongSubscriptions.forEach { start(subscription: $0) }
+        var waitingLongSubscriptions = [RelaySubscription]()
+        var waitingOneTimeSubscriptions = [RelaySubscription]()
+
+        // Counter to track the number of active subscriptions per relay
+        var activeSubscriptionsCount = [URL: Int]()
+
+        all.forEach { relaySubscription in
+            if relaySubscription.isActive {
+                // Update active subscriptions counter
+                let relayAddress = relaySubscription.relayAddress
+                if let currentCount = activeSubscriptionsCount[relayAddress] {
+                    activeSubscriptionsCount[relayAddress] = currentCount + 1
+                } else {
+                    activeSubscriptionsCount[relayAddress] = 1
+                }
+            } else if relaySubscription.isOneTime {
+                waitingOneTimeSubscriptions.append(relaySubscription)
+            } else {
+                waitingLongSubscriptions.append(relaySubscription)
+            }
+        }
+
+        // Start waiting relay subscriptions if they don't exceed the queue
+        // limit
+        let waitingSubscriptions = waitingOneTimeSubscriptions + waitingLongSubscriptions
+        waitingSubscriptions.forEach { relaySubscription in
+            let relayAddress = relaySubscription.relayAddress
+            if let subscriptionsCount = activeSubscriptionsCount[relayAddress] {
+                if subscriptionsCount < queueLimit {
+                    start(subscription: relaySubscription)
+                    activeSubscriptionsCount[relayAddress] = subscriptionsCount + 1
+                }
+            } else {
+                start(subscription: relaySubscription)
+                activeSubscriptionsCount[relayAddress] = 1
+            }
+        }
         
-        if all.count > active.count {
-            Log.debug("\(all.count - active.count) subscriptions waiting in queue.")
+        #if DEBUG
+        let allCount = all.count
+        let activeCount = active.count
```

Comment:
I think I preferred the other way where we didn't have variables for these since this adds two additional lines of code without improving readability.

Reply:
it does not improve readability. However, active is a computed property that iterates over all and filter all those who are active. And the old code used it twice just to print to the console. This way, while a bit weird, doesn't has that performance penalty.